packages/powersync_core/test/in_memory_sync_test.dart

```diff
@@ -312,18 +316,37 @@ void main() {
         },
       );
     });
+
+    test('reconnects when token expires', () async {
```

Comment:
This doesn't actually test the fix because it only affects the channel between the sync worker and the database.

I want to revisit this in the future and run these tests with a fake worker based on a `MessageChannel` in JS. That will take some more work though, so I've added the test now and will do the refactoring in the future.